src/dtgtk/thumbtable.c

```diff
@@ -514,7 +511,7 @@ static void _thumb_move_or_create(dt_thumbtable_t *table,
                                   const gboolean top,
                                   const dt_thumbnail_selection_t sel)
 {
-  if(!th_invalid || g_list_length(*th_invalid)==0)
```

Comment:
This is less readable for sure and saving probably zero of very few cycles.
@ralfbrown : Are you sure this is necessary?

Reply:
I think it is, because otherwise, for each case where we *have* some items in `th_invalid`, we'll traverse all the list items... for nothing !

Reply:
Indeed !